message Strobe {
  required uint32 length = 13;
  required uint32 rate = 12;
  required uint32 on_color = 11;
  optional uint32 off_color = 10;
}

message Fade {
  required uint32 length = 9;
  required uint32 dest_color = 8;
  optional uint32 start_color = 7;
}

message SetColor {
  required uint32 dest_color = 6;
}

message FlickerBaseMessage {
  required uint32 groupid = 1;
  required uint32 timestamp = 2;
  oneof payload {
    Strobe strobe = 3;
    Fade fade = 4;
    SetColor setColor = 5;
  }
}
